src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
It seems to me that `IntegrationFlow` should extend `SmartLifecycle` - I can't think of a scenario where a flow wouldn't need to be so except, perhaps, one that just has a single channel and no endpoints - which is useless.